src/mantine-core/src/components/Modal/Modal.tsx

```diff
@@ -95,6 +98,7 @@ export function MantineModal({
   opened,
   title,
   onClose,
+  onCancel,
```

Comment:
Modal component should not be changed, this logic should be only in `@mantine/modals` package.

Reply:
Looking into this, I'm not sure how this could be done without some changes to the Modal component itself.

This "X" button (and when clicking outside)
![image](https://user-images.githubusercontent.com/9464690/154798132-2fd14f38-53c7-4638-841f-c30bd33c7b05.png)
Directly calls the `closeModal` function from the ctx in the `ModalProvider`. It has no way of knowing about the `onCancel` callback used in the `ConfirmModal` component.

But I will continue and investigate this and see how this could be done from the modals component only